documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3459,4 +3459,61 @@ GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{
 
 **How to fix the violation**: Please add the missing private endpoint API path and operation to the swagger.
 
-Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
\ No newline at end of file
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+
+### <a name="r4037"></a>R4037 MissingTypeObject
+
+**Category** : SDK Error
+
+**Applies to** : ARM and Data plane OpenAPI(swagger) specs
+
+**Output Message** : The schema '{json path}' is considered an object but without a 'type:object', please add the missing 'type:object'.
+
+**Description**: This rule ensure that any schema whose type is 'object' must defines its type explicity, as a schema without type indicates a free-form schema which means it can be any type. For this rule, a schema with keyword 'properties' or 'additionalProperties' is considered an object schema.
+
+**CreatedAt**: May 24, 2021
+
+**LastModifiedAt**: May 24, 2021
+
+**Why this rule is important**: It's used to differentiate free-form object and any type , as the SDK track1 doesn’t make a difference between the two syntax.
```

Comment:
Why this rule is important: It's to differentiate object type from any type in SDK Track 2. While the SDK Track1 generator does not make difference between with and without "type":"object" definition (all treated as "object"), the SDK Track 2 generator requires explicit definition for  object types.